docs/pipelines/yaml-schema.md

```diff
@@ -875,6 +876,83 @@ steps:
 ::: moniker-end
 
 ## Resources
+Any external service that is consumed as part of your pipeline is a resource. An example of a resource can be another CI/CD pipeline that produces artifacts (say Azure pipelines, Jenkins etc.), code repositories (GitHub, Azure Repos, Git), container image registries (ACR, Docker hub etc.).
+
+Resources in YAML represent sources of types pipelines, repositories and containers.
+
+#### Schema
+
+```yaml
+resources:
+  pipelines: [ pipeline ]  
+  repositories: [ repository ]
+  containers: [ container ]
+```
+
+### Pipeline resource
+If you have an Azure Pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. And you can also enable pipeline completion triggers.
```

Comment:
Please put a newline at the end of each sentence. Makes it easier to maintain the doc over time.